### PR TITLE
chore: Add transparent modifier as acceptable for class

### PR DESCRIPTION
**Pull request by tgodzik**
Fixes https://github.com/scalameta/scalameta/issues/3947

I don't think we need to add a dialect for it, since the error isn't really important for parser

### CODE REVIEW

File: scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScannerTokens.scala [hunk 1]
```diff
@@ -123,7 +123,7 @@ final class ScannerTokens(val tokens: Tokens)(implicit dialect: Dialect) {
     }
 
     tokens(index).text match {
-      case soft.KwTransparent() => nextIsDclIntroOrModifierOr(_.is[KwTrait])
+      case soft.KwTransparent() => nextIsDclIntroOrModifierOr(t => t.is[KwTrait] || t.is[KwClass])
```
[the remark this answers is not in the document]
Comment (kitbellew) [reply scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScannerTokens.scala:126]: also, you could use `.isAny[KwTrait, KwClass]` if you'd like.

**Comment by kitbellew** [comment]
> Fixes #3947
> 
> I don't think we need to add a dialect for it, since the error isn't really important for parser

i just checked... apparently, "transparent" is enabled in 3.0 via `allowInlineMods` flag. not sure if that's ok (don't know when `transparent trait` was enabled but `class` only in 3.4, so before that `transparent` was not even a soft keyword), but ok to deal with it separately.

File: scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScannerTokens.scala (hunk 1 above)
[the remark this answers is not in the document]
Comment (tgodzik) [reply scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScannerTokens.scala]: Done! I also checked and object is not valid

**Comment by tgodzik** [comment]
> > Fixes #3947
> > I don't think we need to add a dialect for it, since the error isn't really important for parser
> 
> i just checked... apparently, "transparent" is enabled in 3.0 via `allowInlineMods` flag. not sure if that's ok (don't know when `transparent trait` was enabled but `class` only in 3.4, so before that `transparent` was not even a soft keyword), but ok to deal with it separately.

I could add a separate flag, but I don't really believe this is a big enough change to warrant a separate flag.

**Pull request merged**